core/includes/image.inc

```diff
@@ -526,6 +548,62 @@ function image_hex2rgba(&$hex) {
   );
 }
 
+/**
+ * Get an SVG image's defined dimensions.
+ *
+ * @param string $uri
+ *   The URI or path to an SVG image.
+ *
+ * @return array|FALSE
+ *   An array containing the keys "width" and "height" as integer values. If the
+ *   image is an SVG but no set dimensions are available, these keys will have
+ *   NULL values. If the image is not an SVG, FALSE will be return.
+ */
+function image_get_svg_dimensions($uri) {
+  // Safety check.
+  if (!image_is_svg($uri)) {
+    return FALSE;
+  }
+
+  // Return cached dimensions if already retrieved once this request.
+  $cached_images = &backdrop_static(__FUNCTION__, array());
+  if (isset($cached_images[$uri])) {
+    return $cached_images[$uri];
+  }
+
+  $svg_dimensions = array(
+    'width' => IMAGE_SVG_DEFAULT_WIDTH,
+    'height' => IMAGE_SVG_DEFAULT_HEIGHT,
+  );
+
+  $file_contents = file_get_contents($uri);
+  if ($file_contents) {
+    $svg = simplexml_load_string($file_contents);
```

Comment:
We have to catch here, that simplexml_load_string() returns FALSE on invalid (unparseable) input.
Otherwise the page breaks with `Error: Call to a member function attributes() on bool in image_get_svg_dimensions()`

Reply:
Looks like this is still to do.

Reply:
Done